src/overlays/actors/ovl_En_Raf/z_en_raf.c

```diff
@@ -80,65 +178,787 @@ static DamageTable D_80A1939C = {
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A17C6C.s")
+void EnRaf_ChangeAnimation(EnRaf* this, s32 index) {
+    f32 startFrame = 0.0f;
+    f32 playSpeed = 1.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A17D14.s")
+    this->endFrame = Animation_GetLastFrame(sAnimations[index]);
+    if (index == EN_RAF_ANIMATION_IDLE) {
+        startFrame = this->endFrame;
+    } else if (index == EN_RAF_ANIMATION_CLOSE) {
+        playSpeed = 2.0f;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A17D54.s")
+    Animation_Change(&this->skelAnime, sAnimations[index], playSpeed, startFrame, this->endFrame,
+                     sAnimationModes[index], -4.0f);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A17DDC.s")
+void EnRaf_SetupIdle(EnRaf* this) {
+    Vec3f targetLimbScale = { 1.0f, 1.0f, 1.0f };
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A17E1C.s")
+    EnRaf_ChangeAnimation(this, EN_RAF_ANIMATION_IDLE);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A18080.s")
+    for (i = CARNIVOROUS_LILY_PAD_LIMB_TRAP_1_LOWER_SEGMENT; i <= CARNIVOROUS_LILY_PAD_LIMB_TRAP_3_UPPER_SEGMENT; i++) {
+        Math_Vec3f_Copy(&this->targetLimbScale[i], &targetLimbScale);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A180B4.s")
+    this->petalScaleType = EN_RAF_PETAL_SCALE_TYPE_IDLE_OR_THROW;
+    this->action = EN_RAF_ACTION_IDLE;
+    this->actionFunc = EnRaf_Idle;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/EnRaf_Update.s")
+/**
+ * Sits around waiting for the player or an explosive to get near in order to grab them.
+ */
+void EnRaf_Idle(EnRaf* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    Actor* explosive;
+    f32 xDiff;
+    f32 yDiff;
+    f32 zDiff;
+
+    if (this->timer == 0) {
+        if ((player->transformation != PLAYER_FORM_DEKU) &&
+            (this->dyna.actor.xzDistToPlayer < (BREG(48) + 80.0f) && (player->invincibilityTimer == 0) &&
+             DynaPolyActor_IsInRidingMovingState(&this->dyna) && !(player->stateFlags1 & 0x8000000) &&
+             globalCtx->grabPlayer(globalCtx, player))) {
+            player->actor.parent = &this->dyna.actor;
+            this->grabTarget = EN_RAF_GRAB_TARGET_PLAYER;
+
+            if (player->transformation == PLAYER_FORM_GORON) {
+                this->grabTarget = EN_RAF_GRAB_TARGET_GORON_PLAYER;
+            } else {
+                player->unk_AE8 = 50;
+            }
+
+            this->playerYRotWhenGrabbed = player->actor.world.rot.y;
+            EnRaf_SetupGrab(this);
+            return;
+        }
+
+        if ((globalCtx->gameplayFrames % 2) == 0) {
+            return;
+        }
+
+        explosive = globalCtx->actorCtx.actorLists[ACTORCAT_EXPLOSIVES].first;
+        while (explosive != NULL) {
+            if ((EnRaf*)explosive == this) {
```

Comment:
Well that's weird.

Reply:
Yeah, I have no clue why they did this either. It's definitely some sort of sanity check, but it seems excessive.

Reply:
More than that, it just doesn't make any sense... I'd be tempted to `@bug` it except that it doesn't do anything.

Reply:
Do you want me to slap a comment above this saying it's a pointless check?

Reply:
Yes, that's probably worth doing, if it was totally obvious it wouldn't still be there... I hope.